src/CoreWCF.NetTcp/src/CoreWCF/Channels/Framing/RawStream.cs

```diff
@@ -0,0 +1,264 @@
+﻿using CoreWCF.Runtime;
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CoreWCF.Channels.Framing
+{
+    public class RawStream : Stream
+    {
+        private readonly PipeReader _input;
+        private readonly PipeWriter _output;
+        private bool _canRead;
+        private object _thisLock;
+        private bool _readPending;
+        private TaskCompletionSource<object> _unwrapTcs;
+
+        public RawStream(PipeReader input, PipeWriter output)
+        {
+            _input = input;
+            _output = output;
+            _canRead = true;
+            _thisLock = new object();
+        }
+
+        public override bool CanRead
+        {
```

Comment:
Should not this be used only to say that the Stream supports read operations? IMO Blocking has no place here, but when you do the reading itself, and can read should always return true.